nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -656,37 +705,72 @@ impl BundleCollection {
         Ok(())
     }
 
-    // Perform the work of collecting the support bundle into a temporary directory
-    //
-    // - "dir" is a directory where data can be stored.
-    // - "bundle" is metadata about the bundle being collected.
-    //
-    // If a partial bundle can be collected, it should be returned as
-    // an Ok(SupportBundleCollectionReport). Any failures from this function
-    // will prevent the support bundle from being collected altogether.
-    //
-    // NOTE: The background task infrastructure will periodically check to see
-    // if the bundle has been cancelled by a user while it is being collected.
-    // If that happens, this function will be CANCELLED at an await point.
-    //
-    // As a result, it is important that this function be implemented as
-    // cancel-safe.
-    async fn collect_bundle_as_file(
+    async fn run_collect_bundle_steps(
         self: &Arc<Self>,
-        dir: &Utf8TempDir,
-    ) -> anyhow::Result<SupportBundleCollectionReport> {
-        let log = &self.log;
-
-        info!(&log, "Collecting bundle as local file");
+        output: &Utf8TempDir,
+        mut steps: Vec<(&'static str, CollectionStepFn)>,
+    ) -> SupportBundleCollectionReport {
         let mut report =
             SupportBundleCollectionReport::new(self.bundle.id.into());
 
-        tokio::fs::write(
-            dir.path().join("bundle_id.txt"),
-            self.bundle.id.to_string(),
-        )
-        .await?;
+        const MAX_CONCURRENT_STEPS: usize = 16;
+        let mut tasks =
+            ParallelTaskSet::new_with_parallelism(MAX_CONCURRENT_STEPS);
```

Comment:
Previously, we had some ParallelTaskSets embedded within the collection of sub-pieces of the bundle.

This new "step-based" infrastructure shares that more broadly - everything should be using this one ParallelTaskSet (which is good? that prevents a task that spawns a bunch of other task set, that spawns more task sets - everything is just a unit of work that can get added to this **one** set).